src/components/UploadHistory.vue

```diff
@@ -75,45 +82,15 @@ export default {
     },
     agencyName: function (id) {
       return this.$store.getters.agencyName(id)
-    },
-    loadUploads: async function () {
-      this.uploads = null
-      this.error = null
-
-      const params = new URLSearchParams({ period_id: this.periodId })
-      this.forAgency && params.set('for_agency', this.forAgency)
-      this.only_validated && params.set('only_validated', this.onlyValidated)
-
-      try {
-        const resp = await fetch('/api/uploads?' + params.toString())
-        const result = (await resp.json()) || { error: (await resp.body) }
-
-        if (resp.ok) {
-          this.uploads = result.uploads
-        } else {
-          this.error = `loadUploads API Error (${resp.status}): ${result.error}`
-        }
-      } catch (e) {
-        this.error = `loadUploads Unknown Error: ${e.message}`
-      }
-
-      this.processError()
-    },
-    processError: async function () {
-      if (!this.error) return
-      if (this.onError) {
-        this.onError(this.error)
-        this.error = null
-      }
     }
   },
   watch: {
     periodId: async function () {
-      this.loadUploads()
+      this.$store.dispatch('updateUploads')
     }
   },
   mounted: async function () {
-    this.loadUploads()
+    this.$store.dispatch('updateUploads')
   }
```

Comment:
Thanks, this looks a lot more like I had in mind!

I fully support keeping the "series" and "documents" requests (in Uploads.vue) as local component state.